File: math/9-palindrome-number.js
/*
9. Palindrome Number
https://leetcode.com/problems/palindrome-number/
16 April 2023
David Pratt
*/

/**
 * @param {number} x
 * @return {boolean}
 */
var isPalindrome = function(x) {
    if(x < 0) return false
    let xStr = x.toString()
    
    let revStr = '';
    for(let i = xStr.length - 1; i >= 0; i -= 1){
        revStr += xStr.charAt(i);
    }

    return Number(revStr) === x
};

File: string/9-Palindrome-Number.py
/*
9. Palindrome Number
https://leetcode.com/problems/palindrome-number/
19 November 2023
David Pratt
*/

def reverse(string):
    string = "".join(reversed(string))
    return string
    
class Solution:
    def isPalindrome(self, x: int) -> bool:
        if x < 0:
            return False
        stringX = reverse(str(x))
        return int(stringX) == x
